src/main/java/com/devcycle/sdk/server/local/managers/EventQueueManager.java

```diff
@@ -118,23 +116,15 @@ public void queueAggregateEvent(Event event, BucketedUserConfig bucketedConfig)
     }
 
     private void publishEvents(String sdkKey, FlushPayload flushPayload) throws InterruptedException {
-        Thread publishEventsThread = new Thread(new Runnable() {
```

Comment:
There was no need for this to be in a thread because it is setup and then join() is called which just blocks until the thread completes.  the code can be just be run inline